test/hotspot/jtreg/compiler/cha/TypeProfileFinalMethod.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2021, 2024, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary test c1 to record type profile with CHA optimization
+ * @requires (vm.opt.TieredStoAtLevel == null | vm.opt.TieredStopAtLevel == 4)
+ * @library /test/lib
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ * @run driver compiler.cha.TypeProfileFinalMethod
+ */
+package compiler.cha;
+
+import java.io.File;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.whitebox.WhiteBox;
+
+public class TypeProfileFinalMethod {
+    public static void main(String[] args) throws Exception {
+       ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder(
```

Comment:
The createLimitedTestJavaProcessBuilder ignores any other VM flags. This mode should be used only of test too specific. 
 However, 
 * @requires (vm.opt.TieredStoAtLevel == null | vm.opt.TieredStopAtLevel == 4)
means that we are going to run test with mostly any GC/Runtime and C2 stress options flags. (No C1-only)
So it is needed to use createLimitedTestJavaProcessBuilder to accept all VM flags.
You don't neet to test any additional VM flags until you have a reasons to suppose that something might fails. 
Just use 'createTestJavaProcessBuilder' insted.
If you think that test shouldn't accept any addtional vm flags, use
@requires vm.flagless
instead of
 @requires (vm.opt.TieredStoAtLevel == null | vm.opt.TieredStopAtLevel == 4)
So test is executed only once and don't run if it is gong to ignore flags.

Reply:
Thanks for your suggestions. The test case is dependent on tiered compilation and type profile. Is there any other option for these requirements?

Reply:
I think that additionally 
* @requires vm.opt.TieredCompilation == null | vm.opt.TieredCompilation == true

You could check if test fails with ' -XX:-TieredCompilation'.

Reply:
I tested with "-XX:-TieredCompilation" and it passed.
my test command:
```bash
> make run-test TEST="test/hotspot/jtreg/compiler/cha/TypeProfileFinalMethod.java" CONF=fastdebug JTREG="VM_OPTIONS=-X
X:-TieredCompilation"
...
==============================
Test summary
==============================
   TEST                                              TOTAL  PASS  FAIL ERROR
   jtreg:test/hotspot/jtreg/compiler/cha/TypeProfileFinalMethod.java
                                                         1     1     0     0
==============================
TEST SUCCESS
```

Reply:
Thanks. That's fine for me.